projects/igniteui-angular/src/lib/grids/common/strategy.ts

```diff
@@ -157,7 +157,7 @@ export class IgxSorting implements IGridSortingStrategy {
         const isDate = column?.dataType === DATE_TYPE || column?.dataType === DATE_TIME_TYPE;
         const isTime = column?.dataType === TIME_TYPE;
         const isString = column?.dataType === STRING_TYPE;
-        data = expr.strategy.sort(data, expr.fieldName, expr.dir, expr.ignoreCase, this.getFieldValue, isDate, isTime);
+        data = expr.strategy.sort(data, expr.fieldName, expr.dir, expr.ignoreCase, this.getFieldValue, isDate, isTime, grid);
```

Comment:
What I think we currently lack as an extension point for the strategies (sorting and filtering) is the ability to pass a custom value extraction function as that is currently encapsulated inside the strategy itself (this.getFieldValue). With the pivot we allow setting custom `memberFunction` per dimension, which we currently cannot pass via the public sort/filter methods, hence we use the optional grid parameter and extract the value as needed in the DefaultPivotSortingStrategy/DimensionValuesFilteringStrategy respectively. 

I'm open to the idea of removing the grid parameter as a whole and instead allowing setting of a custom valueExtractor, however that might affect other grids as well (treegrid might be using grid parameter) and would be a bigger change than simply extending the sorting/filtering strategies for the pivot.

I'm open to other ideas as well if you have any.

Reply:
I like the idea to expose _getFieldValue_ or a similar mechanism for the strategies of the grids. I also don't think that falls straight into the scope of this PR. If this can be implemented at a latter point without major user-facing API breakage then this is fine as it is for now.

Reply:
Logged a separate issue here : https://github.com/IgniteUI/igniteui-angular/issues/10922